samza-core/src/main/java/org/apache/samza/config/ApplicationConfig.java

```diff
@@ -104,8 +105,8 @@ public ApplicationMode getAppMode() {
     return Optional.ofNullable(get(APP_MAIN_ARGS));
```

Comment:
Looks like this is unused. Do you where this is used/will be used?
Now that main class is no longer optional, wondering if this should be too?

Reply:
This particular method is not being used as the current toArgs() method is checking the key name directly. We can remove this method.

Reply:
sounds good to me. If you can remove it, that will be great.